test/cliargsparser.cpp

```diff
@@ -406,6 +406,80 @@ TEST_CASE("Sets `execute_cmds` if -x/--execute is provided", "[cliargsparser]")
 	}
 }
 
+TEST_CASE(
```

Comment:
I believe those two test cases should be combined into one, because they check the same thing: the fact that all commands are copied from CLI to `cmds_to_execute`. In other words, they turned out too "low-level", each checking some minute detail. Combining them will also make fail reports nicer, because tests will fail earlier if `cmds_to_execute` is of wrong size (right now, we'll get twice as many errors because both tests will fail separately).

It's a good opportunity to practice working with `std::initializer_list`: the `check` can accept `Opts` and `initializer_list` with expected commands. You can then compare the size of vector to the size of initializer_list. Iterating over them would require writing iterators explicitly, though (or maybe I don't know something—would be glad to learn if you find a solution! Is there `operator==(std::vector<T>, std::initializer_list<T>`? :)